### PR TITLE
feat: support deployment of >1 prefect deployment

### LINKED ISSUE #12: Update custom GHA to support Prefect Projects
currently, it handles only 1 deployment (?)

### PR DESCRIPTION
**Pull request by jamiezieziula**
Resolves #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,12 +62,44 @@ jobs:
...
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Prefect Auth
+        uses: PrefectHQ/actions-prefect-auth@v1
+        with:
+          prefect-api-key: ${{ secrets.PREFECT_API_KEY }}
+          prefect-workspace: ${{ secrets.PREFECT_WORKSPACE }}
```
Comment (parkedwards) [inline README.md:94]: maybe just `...` these steps if they're the same as the single deployment case?

File: README.md [hunk 2]
```diff
@@ -64,12 +62,44 @@ jobs:
...
+
+      - name: Run Prefect Deploy
+        uses: PrefectHQ/actions-prefect-deploy@v3
         with:
-          requirements-file-path: ./examples/simple/requirements.txt
-          entrypoint: ./examples/simple/flow.py:call_api
-          additional-args: --cron '30 19 * * 0'
+          deployment-names: Simple_Deployment_1,Simple_Deployment_2
```
Comment (parkedwards) [inline README.md:99]: im guessing custom actions can't handle arrays as inputs

File: action.yaml [hunk 3]
```diff
@@ -7,36 +7,46 @@ branding:
   color: blue
 
 inputs:
-  additional-args:
+  deployment-names:
     description:
-      Any additinal arguments that need to be passed to
-      the prefect deployment run in a space separated format.
-      i.e - `--variable foo=bar -t 1.0.0 --cron "* * * * *"`.
+      Comma sepearated list of names of your
```
Comment (parkedwards) [inline action.yaml:12]:
```suggestion
      Comma separated list of names of your
```

File: action.yaml [hunk 4]
```diff
@@ -7,36 +7,46 @@ branding:
...
+  requirements-file-paths:
     description:
-      The path to a flow entrypoint within a project,
-      in format `./path/to/file.py:flow_func_name`.
-    required: true
-
-  requirements-file-path:
-    description:
-      Path to requirements files to correctly install
+      Comma sepearated list of paths to
```
Comment (parkedwards) [inline action.yaml:20]:
```suggestion
      Comma separated list of paths to
```

File: README.md (hunk 1 above)
Comment (jamiezieziula) [reply README.md:94]: nice good point

File: README.md (hunk 2 above)
Comment (jamiezieziula) [reply README.md:99]: correct. very annoying

File: action.yaml [hunk 5]
```diff
@@ -7,36 +7,46 @@ branding:
...
     - id: prefect-deploy
       run: |
-        prefect deploy ${{ inputs.entrypoint }} \
-         ${{ inputs.additional-args }}
+        PREFECT_VERSION=$(prefect --version)
+        PREFECT_SHORT_VERSION="$(cut -d "." -f 2 <<< "$PREFECT_VERSION")"."$(cut -d "." -f 3 <<< "$PREFECT_VERSION")"
+        if [ $(echo $PREFECT_SHORT_VERSION'<'10.6 | bc -l) -eq 1 ];
+        then
+          echo "The currently installed Prefect version (2.$PREFECT_SHORT_VERSION) is not compatible. Updating to the latest Prefect version..."
+          pip install prefect -U "prefect>=2.10.8"
```
Comment (parkedwards) [inline action.yaml:46]: i wonder if we just fail here - im sure it's fine, but what if they come with a really old version

File: action.yaml (hunk 4 above)
Comment (jamiezieziula) [reply action.yaml:20]: spelling is hard

**Review by parkedwards — DISMISSED** [review]

File: action.yaml (hunk 5 above)
Comment (jimid27) [reply action.yaml:46]: Yeah I've been back and forth on this - initially I was setting it up to fail but I don't see many (if any) scenarios where a user would go about creating a deployment on like Prefect 2.0.3. We can test the waters with just the prefect install and if we see a submitted issue on this we can re-eval!

**Review by jimid27 — APPROVED** [review]

**Review by parkedwards — APPROVED** [review]

**Pull request merged**